src/utils/guildRole.ts

```diff
@@ -135,3 +142,70 @@ export async function getGuildRoleByName(
   const roles = await getGuildRoles(env);
   return roles?.find((role) => role.name === roleName);
 }
+
+export async function mentionEachUserInMessage({
+  message,
+  userIds,
+  channelId,
+  env,
+}: {
+  message?: string;
+  userIds: string[];
+  channelId: number;
+  env: env;
+}) {
+  const batchSize = 5;
+  let waitTillNextAPICall = 0;
+  try {
+    const failedUsers: Array<string> = [];
+    for (let i = 0; i < userIds.length; i += batchSize) {
+      const batchwiseUserIds = userIds.slice(i, i + batchSize);
+      const messageRequest = batchwiseUserIds.map((userId) => {
+        return fetch(`${DISCORD_BASE_URL}/channels/${channelId}/messages`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bot ${env.DISCORD_TOKEN}`,
+          },
+          body: JSON.stringify({
+            content: `${message ? message + " " : ""} ${userId}`,
+          }),
```

Comment:
NIT: could have been extracted into a util

Reply:
Moving the API call to utils will make things complex. Because we need both raw response and response.json. Also, we are manipulating waitTillNextAPICall through the `.then` block of the fetch call which will become a little complex in case of moving API call into a separate function